app/app/explore/components/TransactionConfirmationModal.tsx

```diff
@@ -69,30 +88,65 @@ export default function TransactionConfirmationModal({
 
         {/* Explorer Links */}
         <div className="w-full space-y-3">
-          <a
-            href={thorchainUrl}
-            target="_blank"
-            rel="noopener noreferrer"
-            className="flex items-center justify-center w-full p-4 bg-white rounded-xl hover:bg-gray-50 transition-colors border text-foreground group"
-          >
-            <span className="mr-2">View on THORChain.net</span>
-            <ExternalLinkIcon />
-          </a>
+          {assetHash && (
+            <>
+              <a
+                href={assetHashThorchainUrl}
+                target="_blank"
+                rel="noopener noreferrer"
+                className="flex items-center justify-center w-full p-4 bg-white rounded-xl hover:bg-gray-50 transition-colors border text-foreground group"
+              >
+                <span className="mr-2">
+                  View {chainName} transaction on THORChain.net
+                </span>
+                <ExternalLinkIcon />
+              </a>
+
+              <a
+                href={assetHashRunescanUrl}
+                target="_blank"
+                rel="noopener noreferrer"
+                className="flex items-center justify-center w-full p-4 bg-white rounded-xl hover:bg-gray-50 transition-colors border text-foreground group"
+              >
+                <span className="mr-2">
+                  View {chainName} transaction on Runescan
+                </span>
+                <ExternalLinkIcon />
+              </a>
+            </>
+          )}
+
+          {runeHash && (
+            <>
+              <a
+                href={runeHashThorchainUrl}
+                target="_blank"
+                rel="noopener noreferrer"
+                className="flex items-center justify-center w-full p-4 bg-white rounded-xl hover:bg-gray-50 transition-colors border text-foreground group"
+              >
+                <span className="mr-2">
+                  View Rune transaction on THORChain.net
+                </span>
+                <ExternalLinkIcon />
+              </a>
 
-          <a
-            href={runescanUrl}
-            target="_blank"
-            rel="noopener noreferrer"
-            className="flex items-center justify-center w-full p-4 bg-white rounded-xl hover:bg-gray-50 transition-colors border text-foreground group"
-          >
-            <span className="mr-2">View on Runescan</span>
-            <ExternalLinkIcon />
-          </a>
+              <a
+                href={runeHashRunescanUrl}
+                target="_blank"
+                rel="noopener noreferrer"
+                className="flex items-center justify-center w-full p-4 bg-white rounded-xl hover:bg-gray-50 transition-colors border text-foreground group"
+              >
+                <span className="mr-2">View Rune transaction on Runescan</span>
+                <ExternalLinkIcon />
+              </a>
```

Comment:
This code and the part that shows the links for the asset seem very similar, could we create a component and use it twice to avoid duplicating code ?